bson/__init__.py

```diff
@@ -351,9 +353,93 @@ def _get_boolean(
     raise InvalidBSON("invalid boolean value: %r" % boolean_byte)
 
 
+class DatetimeMS:
+    """
+    Represents a BSON UTC datetime, which is defined as an int64 of
+    milliseconds since the Unix epoch. Principal use is to represent
+    datetimes outside the range of the Python builtin
+    :class:`~datetime.datetime` class when encoding/decoding BSON.
+    To decode UTC datetimes as a ``DatetimeMS``,
+    `datetime_conversion` in :class:`~bson.CodecOptions` must be set
+    to 'datetime_ms'.
+    """
+
+    def __init__(self, value: Union[int, datetime.datetime]):
+        if isinstance(value, int):
+            self._value = value
+        elif isinstance(value, datetime.datetime):
+            if value.utcoffset() is not None:
+                value = value - value.utcoffset()  # type: ignore
+            self._value = _datetime_to_millis(value)
+        else:
+            raise TypeError(f"{type(value)} is not a valid type for DatetimeMS")
+
+    def __hash__(self) -> int:
+        return hash(self._value)
+
+    def __repr__(self) -> str:
+        return type(self).__name__ + "(" + str(self._value) + ")"
+
+    __str__ = __repr__
+
+    # Avoids using functools.total_ordering for speed.
+
+    def __lt__(self, other: "DatetimeMS") -> bool:
+        return self._value.__lt__(other._value)
```

Comment:
It's a little odd to call `__lt__` (and other "dunder" methods) directly. Let's use <,>,==, etc... instead. EG:
```
        return self._value < other
```